data/apply-planet_osm_point.sql

```diff
@@ -6,6 +6,14 @@ BEGIN
 --------------------------------------------------------------------------------
 
 CREATE INDEX planet_osm_point_place_index ON planet_osm_point(place) WHERE name IS NOT NULL AND place IN ('borough', 'city', 'continent', 'country', 'county', 'district', 'farm', 'hamlet', 'island', 'isolated_dwelling', 'lake', 'locality', 'neighbourhood', 'ocean', 'province', 'quarter', 'sea', 'state', 'suburb', 'town', 'village');
-CREATE INDEX planet_osm_point_level_index ON planet_osm_point(mz_calculate_poi_level("aerialway", "aeroway", "amenity", "barrier", "craft", "highway", "historic", "leisure", "lock", "man_made", "natural", "office", "power", "railway", "shop", "tourism", "waterway")) WHERE mz_calculate_poi_level("aerialway", "aeroway", "amenity", "barrier", "craft", "highway", "historic", "leisure", "lock", "man_made", "natural", "office", "power", "railway", "shop", "tourism", "waterway") IS NOT NULL;
+CREATE INDEX planet_osm_point_level_index ON planet_osm_point(mz_calculate_poi_level("aerialway", "aeroway", "amenity", "barrier", "craft", "highway", "historic", "leisure", "lock", "man_made", "natural", "office", "power", "railway", "shop", "tourism", "waterway", 0::real)) WHERE mz_calculate_poi_level("aerialway", "aeroway", "amenity", "barrier", "craft", "highway", "historic", "leisure", "lock", "man_made", "natural", "office", "power", "railway", "shop", "tourism", "waterway", 0::real) IS NOT NULL;
```

Comment:
Running tests would be good to get some real info.

My gut feeling is that because we've got two effectively unrelated items; the row's geographic position and its `mz_poi_min_zoom`. An index on one has no correlation to the other, and vice-versa. What we want is a combined index where lower values of `mz_poi_min_zoom` will be "higher up" in the tree structure, but (AFAIK) no such index exists.

As I understand it, to do a bitmap index scan would be conceptually similar to:

```SQL
create temporary table geo_rows as
select __row_id__ from table
where way && geo_condition;

create temporary table poi_zoom_rows as
select __row_id__ from table
where poi_min_zoom <= zoom_condition;

select * from table
where __row_id__ in (
  select * from geo_rows union
  select * from poi_zoom_rows);
```

Which means that all the index pages for everything matching `way && geo_condition` *or* `poi_min_zoom <= zoom_condition` need to be loaded. Whereas for one index + filter, it only needs to load one set of index pages, although it'll end up loading more data pages from disk. The optimal is to have a partial index on both conditions... but then we'd be paying the cost on update rather than query.

Hard data for what the actual relative speeds of each are would be very useful! :smile: